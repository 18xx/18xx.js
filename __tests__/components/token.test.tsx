import * as React from 'react';
import { ReactElement } from 'react';
import * as renderer from 'react-test-renderer';

import Token, { TokenProps } from '../../src/components/token';

describe('Token', () => {
  describe('#render()', () => {
    const props: TokenProps = {
      primaryColor: 'red',
      secondaryColor: 'green',
      text: 'ABC',
      textColor: 'blue',
    };

    it ('returns an svg represntation of the token', () => {
      const subject: ReactElement<Token> = (
        <Token {...props} />
      );
      expect(renderer.create(subject)).toMatchSnapshot();
    });

    describe('when the text is faded', () => {
      const subject: ReactElement<Token> = (
        <Token {...props} faded={true} />
      );
      expect(renderer.create(subject)).toMatchSnapshot();
    });

    describe('when the text is a number', () => {
      it ('returns an svg represntation of the token', () => {
        const subject: ReactElement<Token> = (
          <Token {...props} text='2' />
        );
        expect(renderer.create(subject)).toMatchSnapshot();
      });
    });
  });
});
