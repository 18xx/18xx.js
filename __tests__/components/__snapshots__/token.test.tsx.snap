exports[`Token #render() returns an svg represntation of the token 1`] = `
<svg
  height="40"
  opacity={1}
  width="40">
  <circle
    cx={20}
    cy={20}
    fill="red"
    r={16} />
  <rect
    fill="green"
    height={12}
    rx={1.75}
    ry={12}
    width={32}
    x={4}
    y={14} />
  <text
    fill="blue"
    fontSize={10}
    fontWeight="normal"
    stroke="none"
    textAnchor="middle"
    x={20}
    y={24}>
    ABC
  </text>
</svg>
`;
